python/lib/core/dmod/test/test_context.py

```diff
@@ -0,0 +1,1599 @@
+"""
+Unit tests used to ensure that dmod.core.context operations behave as intended
+"""
+from __future__ import annotations
+
+import abc
+import dataclasses
+import inspect
+import os
+import sys
+import typing
+import unittest
+import multiprocessing
+import string
+import random
+import re
+
+from collections import namedtuple
+from itertools import permutations
+from multiprocessing import managers
+from multiprocessing import pool
+
+from typing_extensions import ParamSpec
+from typing_extensions import Self
+
+from ..core import context
+
+VARIABLE_ARGUMENTS = ParamSpec("VARIABLE_ARGUMENTS")
+
+T = typing.TypeVar("T")
+R = typing.TypeVar("R")
+
+STATIC_METHOD_ONE_VALUE = 95
+STATIC_METHOD_TWO_VALUE = "blue"
+
+CLASS_METHOD_ONE_VALUE = False
+CLASS_METHOD_TWO_VALUE = 3+3j
+
+STRING_INTEGER_PATTERN = re.compile(r"-?\d+")
+MEMBER_SPLIT_PATTERN = re.compile(r"[./]")
+
+MutationTuple = namedtuple("MutationTuple", ["field", "value", "should_be_equal"])
+
+
+def shared_class_two_instance_method_formula(*args) -> int:
+    """
+    The formula for the instance method for SharedClassTwo
+
+    Args:
+        *args: The values to include in the formula
+
+    Returns:
+        An integer representing the size of passed in values
+    """
+    total: int = 0
+
+    for item in args:
+        if isinstance(item, typing.Sized):
+            total += len(item)
+        else:
+            total += item
+
+    return total
+
+
+def make_word(min_length: int = None, max_length: int = None, character_set: str = None, avoid: str = None) -> str:
+    """
+    Create a random jumble of characters to build a new word
+
+    Args:
+        min_length: The shortest the word can be
+        max_length: The longest a word can be
+        character_set: What characters can make up the word
+        avoid: A word to avoid creating
+
+    Returns:
+        A semi random string of a semi-random length
+    """
+    if min_length is None:
```

Comment:
Just make these the default values for the parameters pls. That is way clearer.

Reply:
Done